app/webpacker/components/CompetitionsOverview/CompetitionsFilters.js

```diff
@@ -88,7 +88,16 @@ export function EventSelector({
       <label htmlFor="events">
         {`${I18n.t('competitions.competition_form.events')}`}
         <br />
-        <Button disabled={disabled || eventList.length >= maxEvents} primary type="button" size="mini" id="select-all-events" onClick={() => onEventSelection({ type: 'select_all_events' })}>{I18n.t('competitions.index.all_events')}</Button>
+        <Popup
+          disabled={!Number.isFinite(maxEvents)}
+          trigger={
+            <span><Button disabled={disabled || eventList.length >= maxEvents} primary type="button" size="mini" id="select-all-events" onClick={() => onEventSelection({ type: 'select_all_events' })}>{I18n.t('competitions.index.all_events')}</Button></span>
```

Comment:
Why are you wrapping the button into a `span`?
And if the button is sometimes disabled, then shouldn't the whole popup also be disabled in the first place? Or will SemUI detect this automatically for you?

Reply:
because of a bug that disabled buttons do not probive popups. I am setting the popup to disabled on `disabled={!Number.isFinite(maxEvents)}`